mempool/src/tests/mocks.rs

```diff
@@ -67,14 +102,8 @@ impl MockSharedMempool {
         let network_events = MempoolNetworkEvents::new(network_notifs_rx, conn_notifs_rx);
         let (ac_client, client_events) = mpsc::channel(1_024);
         let (consensus_sender, consensus_events) = mpsc::channel(1_024);
-        let (mempool_notifier, mempool_listener) = match mempool_listener {
-            None => {
-                let (mempool_notifier, mempool_listener) =
-                    mempool_notifications::new_mempool_notifier_listener_pair();
-                (Some(mempool_notifier), mempool_listener)
-            }
-            Some(mempool_listener) => (None, mempool_listener),
-        };
+        let (mempool_notifier, mempool_listener) =
+            mempool_notifications::new_mempool_notifier_listener_pair();
```

Comment:
Interesting, was a listener never passed in?

Reply:
The listener was not required in some tests. This change makes it so that we're passing in a listener. IIRC, I don't think it should have any effect on the unit tests 😄

Reply:
listener is stored in the mock object, outside can access it without passing in one, thus removed input mempool_listener and changed to use mock#mempool_listener in the place passed in a mempool_listener.